components/TOC.js

```diff
@@ -20,10 +20,20 @@ export default function TOC({
     slugWithATag: item.content.replace(/<|>|"|\\|\/|=/gi, '').replace(/\s/gi, '-').toLowerCase()
 }))
   const [open, setOpen] = useState(false)
+  
+  useEffect(() => {
+    const navbar = document.getElementById("navbar");
+    const toc= document.querySelector(".toc");
+    const height = (navbar.offsetHeight).toString(); 
+  
+    toc.style.setProperty("max-height", `calc(100vh - ${height}px)`)
+    toc.style.setProperty(`${cssBreakingPoint}:top`, `${height}px`)
+  }, []);
```

Comment:
I'm usually not in favor of DOM manipulations in React. Isn't there a way we can achieve the same without accessing the DOM? Also, what's this piece of code trying to do? Would be great to have a comment explaining the purpose of it.

Reply:
Sorry @fmvilas, I and @mcturco discussed a lot about this error on my previous PRs and Slack. We couldn't able to get to the perfect way, other than this. Although, you may think that they are a bit unusual from other parts of code, but it is actually improving the stability issue in a better way. 
Yepp, that's a good suggestion that I should add comments in the code to explain the approach used.

Reply:
> Since the header height could change in the future, we have to get the height dynamically instead of creating a class and giving it a px value.

The fact that the header height can change in a future design doesn't mean we have to get this value dynamically IMHO. In such a case, I think we can just change [this line](https://github.com/asyncapi/website/blob/master/tailwind.config.js#L102) to use a variable called `headerHeight` (instead of a hardcoded `4rem`). We also add a class after [this one](https://github.com/asyncapi/website/blob/master/tailwind.config.js#L83) with the same value and use it on the NavBar component. This way, when we change it in the future, everything will automatically align too.

Why I'm advocating against this? You may wonder. Well, what we're doing here is a mix of CSS architecture paradigms. We're mixing utility-first (Tailwind) with a sort of BEM where we assign a class with the name of the component (`toc`, `sidebar`, etc.) It may seem innocent right now but it's a hack. And it's a hack today, another one tomorrow, and so on, and then one day the codebase is a mess because we didn't stick to a specific pattern.

There are ways to do this in a utility-first approach like I described above. And this change is not "necessary". It's something that seems to be an improvement at first but I don't think it will be in the long term. We're violating a few principles just for this:

1. Accessing the DOM, going against React best practices.
2. Mixing utility-first with other CSS architecture paradigms.
3. Assigning ids to elements, breaking reusability.

Again, I wouldn't be worried if that's something we have to do and there's no other way but that's not the case, the ToC is there looking good and there's an alternative way of doing it.

Reply:
I mean, the idea is good. Don't get me wrong. I just don't think we should be accessing DOM and doing other hacks. My comment was more directed at @akshatnema, sorry for the confusion 😄

Reply:
Yeah only reason why I commented was because it was my suggestion to @akshatnema in the first place, so just wanted to apologize for steering in the wrong direction

Reply:
No problem. It's ok to discuss stuff. No need to apologize :)

Reply:
@fmvilas Yepp, I agree that the codebase has used utility-first approach and I should use inbuilt CSS classes only to customize every component. But, how can I use Navbar height dynamically on each screen size so as to provide correct parameters to `toc`. If we make the Navbar of fixed size for every screen, that's not the correct way to do it because it can give me responsiveness issues on various screens. Also, how will we pass `Navbar` height to the tailwind.css.config function parameters. Is there a way to make customize class in tailwind config file such that it takes customized height parameter in px and then it will return the CSS properties of required height and properties?

This probably seems very confusing and complicated to implement it using `tailwind.config.css` and then adding class to the component. If you have a plan to do so, kindly guide me through your approach.

Reply:
Yeah, the Tailwind config file let's you create custom classes (see the links I shared for examples). You can create multiple navbar height values for different screen sizes and then apply them on navbar component using syntax like `lg:navbar-height`, `sm:navbar-height`, etc. Does it make sense? I think the height is always 4rem except on mobile but better check.

Then for ToC you can edit the height values on Tailwind config file (see link that I shared before). Not sure I'm explaining myself 😅 

Happy to help you on Slack next week. I'm going on holidays after this message 😆🍺

Reply:
Yepp, I got the way you want. I and Missy have already discussed this before in Slack. But, what I faced problems in your approach are these:
- If we hardcoded these `top` and `max-height` attributes, it is sure that we have to consider each screen and many classes will then be added in a single div, so as it gets suited everywhere. This possibly can create a mess in the className if somebody else (or beginner) comes to see it. He has to take more time in understanding the situation in a better way. Also, it is not a good practice to overlap the attributes again and again on the same element.

- If we still neglect the above situation and go on providing different `top` and `max-height` for different screen sizes, there is a possibility in the future that the design of `Navbar` or `TOC` or `Sidebar` may change. Then, the height of one of the components may get changed. In this case, again we have to reconsider the above attributes in providing accurate units to them.

Considering the above 2 problems, @mcturco suggested me to have the attributes assigned in such a way that at whatever times, `Navbar` or `TOC` or `Sidebar` changes, it should not affect the responsive and alignment issues while making them `sticky`. So, we came to this solution which dynamically assigns `top` and `max-height` parameters according to the current height of Navbar in that particular screen. Even, if the new styles will be implemented on the website, this implementation won't be making any issues that are right now on the website, related to these components.

Sorry for writing so much in the message :smile:, I want to explain my point of view what I faced the difficulties while implementing with your approach. Meanwhile, @fmvilas, you can surely enjoy your holidays. Hope you have great fun :+1:. I will ask other maintainers too for their review on this.

Reply:
>The problem I see, which should be fixed, is that the height is calculated (and should be added by css class) only during component initialization (and that it never removes) so if someone rotates the screen on a smartphone from vertical to horizontal view then may have the wrong height.

As far I saw the preview in my phone and my codebase, I haven't changed anything in the mobile view yet. So, there is no issue if someone rotates the screen on a smartphone from vertical to horizontal view. 

>I think we can't do without JS here, but instead of replacing the height manually we should have a class (or several if needed, like Fran said sm:... etc) that would be added to the prop className and this class should be changed when there is an event. I hope you understand :)

This can be done, although it will take some time. I have a doubt regarding `tailwind.config.js` that if I add some more classes in it and export it globally, it doesn't get into the production URL. Do I need to run some commands to recompile CSS for the updated `tailwind.config.js` file? I really need it for this and other PRs. I expect an answer from you @magicmatatjahu :)

Reply:
I'm back from holidays. Happy to do a pair-programming session with you @akshatnema if needed. Ping me on Slack if you want. The solution I'm proposing is really simple and straightforward, you'll see :)

Reply:
@fmvilas Sure, I will love to do a pair-programming session with you and your solution to the problem. Sorry for responding late to the message, I was busy with my own academics today 😞. Meanwhile, I will connect you in Slack to better communicate with you. Just small permission, should I dm you or ask in the desired channel of slack?

Reply:
No rush mate :) Just ping me directly or use the tooling channel if you prefer. As you wish. We can also live stream the session if you want.